internal/collectors/collectors.go

```diff
@@ -35,19 +36,31 @@ const (
 var NoResult = Result{}
 
 type Collector interface {
-	run(arch Architecture) (Result, error)
+	run(arch string) (Result, error)
 }
 
-func CollectInformation(architecture Architecture, collectors []Collector) (Result, error) {
+func CollectInformation(architecture string, collectors []Collector) (Result, error) {
 	obj := Result{}
 
 	for _, collector := range collectors {
 		res, err := collector.run(architecture)
 		if err != nil {
-			return NoResult, err
+			util.Debug.Printf("Collecting system information failed: %s", err)
 		}
 		maps.Copy(obj, res)
 	}
 
 	return obj, nil
 }
+
+func FromResult[R any](result Result, key string, def R) R {
```

Comment:
Oh, generics, fancy 😉 ! Could you document this?

Reply:
Added a comment about what it does